cds_sorenson/api.py

```diff
@@ -122,35 +130,76 @@ def get_encoding_status(job_id):
     job_status = status_json.get('Status', {}).get('Status')
     job_progress = status_json.get('Status', {}).get('Progress')
     if job_status:
-        return SORENSON_STATUSES.get(job_status), job_progress
+        return statuses.get(job_status), job_progress
     # status not found? check in different place
     job_status = status_json.get('StatusStateId')
     if job_status:
         # job is probably either finished or failed, so the progress will
         # always be 100% in this case
-        return SORENSON_STATUSES.get(job_status), 100
+        return statuses.get(job_status), 100
     # No status was found (which shouldn't happen)
     raise SorensonError('No status found for job: {0}'.format(job_id))
 
 
-def restart_encoding(job_id, input_file, preset_ID, output_file=None):
+def restart_encoding(job_id, input_file, output_file, preset_quality,
+                     display_aspect_ratio, **kwargs):
     """Try to stop the encoding job and start a new one.
 
-    It's impossible to get the input_file and preset_ID from the job_id, if
-    the job has not yet finished, so we need to specify all parameters for
-    stopping and starting the encoding job.
+    It's impossible to get the input_file and preset_quality from the
+    job_id, if the job has not yet finished, so we need to specify all
+    parameters for stopping and starting the encoding job.
     """
     try:
         stop_encoding(job_id)
     except SorensonError:
         # If we failed to stop the encoding job, ignore it - in the worst
         # case the encoding will finish and we will overwrite the file.
         pass
-    return start_encoding(input_file, preset_ID, output_file)
+    return start_encoding(input_file, output_file, preset_quality,
+                          display_aspect_ratio, **kwargs)
 
 
 def get_presets_by_aspect_ratio(aspect_ratio):
     """Return the list of preset IDs for a given aspect ratio."""
-    return [preset.get('preset_id')
-            for preset in current_app.config['CDS_SORENSON_PRESETS']
-                                     .get(aspect_ratio, [])]
+    try:
+        inner_dict = current_app.config['CDS_SORENSON_PRESETS'][aspect_ratio]
+        return [preset['preset_id'] for preset in inner_dict.values()]
+    except KeyError:
+        raise InvalidAspectRatioError(aspect_ratio)
+
+
+def get_available_aspect_ratios(pairs=False):
+    """Return all available aspect ratios.
+
+    :param pairs: if True, will return aspect ratios as pairs of integers
+    """
+    ratios = [key for key in current_app.config['CDS_SORENSON_PRESETS']]
+    if pairs:
+        ratios = [tuple(map(int, ratio.split(':', 1))) for ratio in ratios]
+    return ratios
+
+
+def get_available_preset_qualities():
+    """Return all available preset qualities."""
+    return list(OrderedDict.fromkeys(chain(*[
+        outer_dict.keys()
+        for outer_dict in current_app.config['CDS_SORENSON_PRESETS'].values()
+    ])))
```

Comment:
Can you please split this kind of stuff for a better readability for people like me, who have problems with understanding what's the result of those 6 commands chained together on 4 lines of code ? :wink: 
Variables are cheap.
```
all_qualities = [presets.keys()
    for presets in current_app.config['CDS_SORENSON_PRESETS'].values()]

return list(OrderedDict.fromkeys(chain(*all_qualities)))
```